kvscheduler/api/errors.go

```diff
@@ -62,3 +73,16 @@ func ErrInvalidMetadataType(key string) error {
 	}
 	return fmt.Errorf("metadata has invalid type for key: %s", key)
 }
+
+// NonRetriableIfInTheList returns callback suitable for KVDescriptor.RetriableFailure
+// based on an exhaustive list of all possible non-retriable errors.
+func NonRetriableIfInTheList(errorList []error) func(err error) bool {
+	return func(err error) bool {
+		for _, nonRetriableError := range errorList {
+			if err.Error() == nonRetriableError.Error() {
```

Comment:
Not sure what is the intended use case for this function, since it's used internally only and the loop over the errors can be done there. Also you should compare the types directly, not the string output from `Error`.

Reply:
It is helper for descriptors to generate callback for:
`RetriableFailure func(err error) bool`

But perhaps instead of the callback it could be just a list of non-retriable errors. Would it be sufficient for any descriptor?

Reply:
I would keep the way it is now and let each descriptor handle thier errors inside `RetriableFailure` function defined in Descriptor on their own and just get rid of this helper method, don't be afraid to be verbose here. Better to keep the logic of deciding which error is retriable inside the function in Descriptor.